src/analysis/mean_inequalities.lean

```diff
@@ -522,3 +523,169 @@ begin
+lemma lintegral_mul_le_of_ne_zero_of_ne_top {p q : ℝ} (hpq : p.is_conjugate_exponent q)
+  {f g : α → ennreal} (hf : measurable f) (hg : measurable g)
+  (hf_nontop : ∫⁻ a, (f a)^p ∂μ ≠ ⊤) (hg_nontop : ∫⁻ a, (g a)^q ∂μ ≠ ⊤)
+  (hf_nonzero : ∫⁻ a, (f a)^p ∂μ ≠ 0) (hg_nonzero : ∫⁻ a, (g a)^q ∂μ ≠ 0) :
+  ∫⁻ a, (f * g) a ∂μ ≤ (∫⁻ a, (f a)^p ∂μ)^(1/p) * (∫⁻ a, (g a)^q ∂μ)^(1/q) :=
+begin
+  let npf := (∫⁻ (c : α), (f c) ^ p ∂μ) ^ (1/p),
+  let nqg := (∫⁻ (c : α), (g c) ^ q ∂μ) ^ (1/q),
+  calc ∫⁻ (a : α), (f * g) a ∂μ
+    = ∫⁻ (a : α), ((fun_mul_inv_snorm f p μ * fun_mul_inv_snorm g q μ) a)
+      * (npf * nqg) ∂μ :
+  begin
+    refine lintegral_congr (λ a, _),
+    rw [pi.mul_apply, fun_eq_fun_mul_inv_snorm_mul_snorm f hf_nonzero hf_nontop,
+      fun_eq_fun_mul_inv_snorm_mul_snorm g hg_nonzero hg_nontop, pi.mul_apply],
+    ring,
+  end
+  ... ≤ npf * nqg :
+  begin
+    rw lintegral_mul_const' (npf * nqg) _ (by simp [hf_nontop, hg_nontop, hf_nonzero, hg_nonzero]),
+    nth_rewrite 1 ←one_mul (npf * nqg),
+    refine mul_le_mul _ (le_refl (npf * nqg)),
+    have hf1 := lintegral_rpow_fun_mul_inv_snorm_eq_one hpq.pos hf hf_nonzero hf_nontop,
+    have hg1 := lintegral_rpow_fun_mul_inv_snorm_eq_one hpq.symm.pos hg hg_nonzero hg_nontop,
+    exact lintegral_mul_le_one_of_lintegral_rpow_eq_one hpq (hf.ennreal_mul measurable_const)
+      (hg.ennreal_mul measurable_const) hf1 hg1,
+  end
+end
+
+lemma ae_eq_zero_of_lintegral_rpow_eq_zero {p : ℝ} (hp0_lt : 0 < p) {f : α → ennreal}
+  (hf : measurable f) (hf_zero : ∫⁻ a, (f a)^p ∂μ = 0) :
+  f =ᵐ[μ] 0 :=
+begin
+  rw lintegral_eq_zero_iff hf.ennreal_rpow_const at hf_zero,
+  refine filter.eventually.mp hf_zero (filter.eventually_of_forall (λ x, _)),
+  dsimp only,
+  rw [pi.zero_apply, rpow_eq_zero_iff],
+  intro hx,
+  cases hx,
+  { exact hx.left, },
+  { exfalso,
+    linarith, },
+end
+
+lemma lintegral_mul_eq_zero_of_lintegral_rpow_eq_zero {p : ℝ} (hp0_lt : 0 < p)
+  {f g : α → ennreal} (hf : measurable f) (hf_zero : ∫⁻ a, (f a)^p ∂μ = 0) :
+  ∫⁻ a, (f * g) a ∂μ = 0 :=
+begin
+  rw ←@lintegral_zero_fun α _ μ,
+  refine lintegral_congr_ae _,
+  suffices h_mul_zero : f * g =ᵐ[μ] 0 * g , by rwa zero_mul at h_mul_zero,
+  have hf_eq_zero : f =ᵐ[μ] 0, from ae_eq_zero_of_lintegral_rpow_eq_zero hp0_lt hf hf_zero,
+  exact filter.eventually_eq.mul hf_eq_zero (ae_eq_refl g),
+end
+
+lemma lintegral_mul_le_of_ne_zero_of_eq_top {p q : ℝ} (hp0_lt : 0 < p) (hq0 : 0 ≤ q)
+  {f g : α → ennreal} (hf_top : ∫⁻ a, (f a)^p ∂μ = ⊤) (hg_nonzero : ∫⁻ a, (g a)^q ∂μ ≠ 0) :
+  ∫⁻ a, (f * g) a ∂μ ≤ (∫⁻ a, (f a)^p ∂μ) ^ (1/p) * (∫⁻ a, (g a)^q ∂μ) ^ (1/q) :=
+begin
+  refine le_trans le_top (le_of_eq _),
+  have hp0_inv_lt : 0 < 1/p, by simp [hp0_lt],
+  rw [hf_top, ennreal.top_rpow_of_pos hp0_inv_lt],
+  simp [hq0, hg_nonzero],
+end
+
+/-- Hölder's inequality for functions `α → ennreal`. -/
+lemma lintegral_mul_le (μ : measure α) {p q : ℝ} (hpq : p.is_conjugate_exponent q)
```

Comment:
In the finite sum case, the name of the theorem is `inner_le_Lp_mul_Lq`. For coherence, you could maybe name your lemma `lintegral_mul_le_Lp_mul_Lp` (and adapt similarly the names of previous statements). And also, it can be a theorem, not just a lemma! Also, you can expand a little bit the docstring (see the finite sum one).